test/functional/test_runner.py

```diff
@@ -285,11 +285,12 @@ def main():
 
     # Remove the test cases that the user has explicitly asked to exclude.
     if args.exclude:
-        exclude_tests = [re.sub("\.py$", "", test) + (".py" if ".py" not in test else "") for test in args.exclude.split(',')]
+        exclude_tests = [re.sub("\.py$", "", test) for test in args.exclude.split(',')]
```

Comment:
nit: could remove the use of the `re` module, since it is no longer required? `"foo.py".split('.py')[0] == "foo".split('.py')[0]`